tests/accuracy/test_lhc_rdts.py

```diff
@@ -0,0 +1,168 @@
+    """
+    paths = {}
+    for beam in (1, 2):
+        path = tmp_path_factory.mktemp(f"analysis_beam{beam}", numbered=False)
+        paths[beam] = path
+    return paths
+
+
+@pytest.fixture(scope="module", autouse=True)
+def run_selective_harpy():
+    """Run the harpy analysis for selected RDTs.
+
+    This fixture runs the harpy analysis for beam 1 and beam 2 to test the
+    opposite_direction flag in combination with optics=True.
+    """
+    run_harpy(beam=1)
+    run_harpy(beam=2)
+    yield # Run the tests
+
+    # Clean up the analysis files
+    for analysis_path in FREQ_OUT_DIR.iterdir():
+        analysis_path.unlink()
+    FREQ_OUT_DIR.rmdir()
+
+
+AMPLITUDE_TOLERANCES = {
+    1: {  # Beam 1: 0.6% and 4% for sextupoles and octupoles respectively
+        "sextupole": 6e-3,
+        "octupole": 4e-2,
+    },
+    2: {  # Beam 2: 0.6% and 5% for sextupoles and octupoles respectively
+        "sextupole": 6e-3,
+        "octupole": 5e-2,
+    },
+}
+PHASE_TOLERANCES = {
+    1: {  # Beam 1: 0.04 rad and 0.03 rad for orders 2 and 3 respectively
+        "sextupole": 4e-2,
+        "octupole": 3e-2,
+    },
+    2: {  # Beam 2: 0.07 rad and 0.05 rad for orders 2 and 3 respectively
+        "sextupole": 7e-2,
+        "octupole": 5e-2,
+    },
+}
+
+
+@pytest.mark.parametrize("rdt", get_rdts())
+@pytest.mark.parametrize("beam", [1, 2], ids=lambda x: f"Beam{x}")
+def test_lhc_rdts(beam: int, rdt: str, initialise_test_paths):
+    """Test the RDTs calculated by OMC3 against the analytical model and MAD-NG.
+
+    The test forces harpy to run for the selected RDTs to ensure the use of the
+    opposite_direction flag is consistent with the optics=True, beam=2 situation.
+
+    The configuration for the test is set in the file tests/inputs/lhc_rdts/rdt_constants.py:
+    - NTURNS = 1000
+    - KICK_AMP = 1e-3
+    - SEXTUPOLE_STRENGTH = 3e-5
+    - OCTUPOLE_STRENGTH = 3e-3
+
+    If this configuration is changed, the test needs to be updated accordingly by running
+    the create_data.py script in the same directory.
+
+    This configuration was specifically chosen to ensure that OMC3 is given a fair chance
+    to produce correct results. The main issues that can arise is detuning. The octupole
+    and sextupole strengths are chosen to be small enough to avoid large detuning effects,
+    but also large enough to produce a measurable effect. Furthermore, the larger each
+    strength, the more likely the regime will become nonlinear, which can lead to larger
+    errors in the RDT calculations.
+    """
+    # Retrieve the current RDT configuration
+    _, order = get_rdt_type(rdt)
+
+    # Get the temporary analysis path
+    analysis_path = initialise_test_paths[beam]
+
+    # Now perform the optics calculations from the harpy output
+    # As this is run per RDT, the check_previous is set to True to avoid rerunning the analysis
+    rdt_dfs = get_rdts_from_harpy(beam, output_dir=analysis_path, check_previous=True)
```

Comment:
to avoid re-running things, instead of the `check_previous` this could also be a fixture with scope="module"